magenta-lib/src/main/scala/magenta/deployment_type/CloudFormationDeploymentTypeParameters.scala

```diff
@@ -104,6 +108,24 @@ trait CloudFormationDeploymentTypeParameters {
       """.stripMargin
   ).default(false)
 
+  val minInstancesInServiceParameters = Param[Map[CfnParam, TagCriteria]](
+    "minInstancesInServiceParameters",
+    optional = true,
+    documentation = """Mapping between a CloudFormation parameter controlling the MinInstancesInService property of an ASG UpdatePolicy and the ASG.
+        |
+        |For example:
+        |```
+        |  minInstancesInServiceParameters:
+        |    MinInstancesInServiceForApi:
+        |      App: my-api
+        |    MinInstancesInServiceForFrontend:
```

Comment:
I think this is a bit optimistic! `MinInstancesInServiceForFrontend` looks nice, but I think it is actually going to be `MinInstancesInServiceForfrontend` (note the lower case `f`). This is even more confusing when we don't have nice short names like "Frontend" and "Api". I was wondering whether we should add a delimiter to the name?

`MinInstancesInServiceFor-frontend` is probably easier to parse at a glance? Probably need to think about this alongside the CDK change as well, so not something to consider in isolation here!

Reply:
GuCDK's logic is here - https://github.com/guardian/cdk/pull/2476/files#diff-60795fa553ec02295a0718711b79382e5aa10022970df8240822e4a214fe114dR130-R132.

It's worth noting that these map to CFN parameters in the template; `-` are not allowed in CFN parameters.

Reply:
I checked the tags restrictions but not the parameter name ones, sorry! If those are just alphanumerics then this is a bit harder 😅 

No dramas!